core-bundle/src/DependencyInjection/ContaoCoreExtension.php

```diff
@@ -281,27 +282,25 @@ private function handleTokenCheckerConfig(array $config, ContainerBuilder $conta
         }
     }
 
-    private function handleLegacyRouting(array $config, ContainerBuilder $container, YamlFileLoader $loader): void
+    private function handleLegacyRouting(array $mergedConfig, array $configs, ContainerBuilder $container, YamlFileLoader $loader): void
     {
-        $count = 0;
-
-        if (!isset($config['prepend_locale'])) {
-            ++$count;
-            $config['prepend_locale'] = false;
-        }
-
-        if (!isset($config['url_suffix'])) {
-            ++$count;
-            $config['url_suffix'] = '.html';
+        if (false === $mergedConfig['legacy_routing']) {
+            foreach ($configs as $config) {
+                if (isset($config['prepend_locale'])) {
+                    throw new InvalidConfigurationException('Configuring contao.prepend_locale to "'.var_export($config['prepend_locale'], true).'" requires legacy routing.');
```

Comment:
```suggestion
                    throw new InvalidConfigurationException('Setting contao.prepend_locale to "'.var_export($config['prepend_locale'], true).'" requires legacy routing.');
```

Reply:
Why is it important to show the configured value in the exception message? `Setting contao.prepend_locale to "false" requires legacy routing.` somehow implies that setting it to `true` does not.

Reply:
It is not important, but I thought it would give a hint to what was configured.